### PR TITLE
Run upgrade tests if they have changed

### PR DESCRIPTION
**Pull request by quodlibetor**

### CODE REVIEW

**Review by benesch — CHANGES_REQUESTED** [review]
This will run upgrade tests on every PR. Possibly that is what we want, but if so then they don’t need to be run nightly.

**Comment by quodlibetor** [comment]
sure if you'd prefer a blocklist to an allowlist I'll do that.

**Comment by benesch** [comment]
Ah sorry, I was unclear, but putting this in the plugin config doesn't make sense because it's not actually part of the plugin. That `exclude_image_dependencies` should be part of the top-level pipeline config so that it applies regardless of the plugin.

**Comment by quodlibetor** [comment]
I believe that we aren't allowed to add top-level keys, based on the way that [this test failed](https://buildkite.com/materialize/tests/builds/19887#6941b4d1-f845-4c26-97ba-30760a91461c). I couldn't find any docs about adding keys to top-level config items, but obviously I could have missed it, not super familiar with buildkite.

**Comment by benesch** [comment]
Right, you'll need to delete it from the config like we do for the `inputs` key: https://github.com/MaterializeInc/materialize/blob/3fe93a5aa8d56219c3d90dfbaf147efcba270188/ci/test/mkpipeline.py#L66-L70

**Comment by quodlibetor** [comment]
oh, duh.

File: ci/test/mkpipeline.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ class PipelineStep:
     def __init__(self, id: str):
         self.id = id
         self.extra_inputs: Set[str] = set()
+        self.only_inputs: Set[str] = set()
```
Comment (benesch) [inline ci/test/mkpipeline.py:84]: This looks leftover.

**Review by benesch — APPROVED** [review]
LGTM modulo the leftover var. Thanks!

**Review by benesch — APPROVED** [review]

**Pull request merged**